bionic/deriver.py

```diff
@@ -181,47 +184,48 @@ def _initialize_task_state(self, task_state):
 
         # All names in this task should point to the same provider.
         task_state.provider, = set(
-            self._flow_state.get_provider(task_key.entity_name)
+            self._flow_state.get_provider(task_key.dnode.to_entity_name())
             for task_key in task.keys
         )
         # And all the task keys should have the same case key.
         task_state.case_key, = set(task_key.case_key for task_key in task.keys)
 
         task_state.is_initialized = True
 
-    def _populate_entity_info(self, entity_name):
-        if entity_name in self._task_lists_by_entity_name:
+    def _populate_dnode_info(self, dnode):
+        if dnode in self._task_lists_by_dnode:
             return
 
+        entity_name = dnode.to_descriptor()
         provider = self._flow_state.get_provider(entity_name)
 
         dep_names = provider.get_dependency_names()
         for dep_name in dep_names:
-            self._populate_entity_info(dep_name)
+            self._populate_dnode_info(DescriptorNode.from_descriptor(dep_name))
 
         dep_key_spaces_by_name = {
-            dep_name: self._key_spaces_by_entity_name[dep_name]
+            dep_name: self._key_spaces_by_dnode[
```

Comment:
same here, why name and not dnode? Actually, same comment for all other maps in this PR too.